src/platform/plugins/shared/workflows_execution_engine/server/plugin.ts

```diff
@@ -341,69 +392,77 @@ export class WorkflowsExecutionEnginePlugin
-            id: workflowExecution.id,
-            status: ExecutionStatus.FAILED,
-            error: {
-              type: 'Error',
-              message: 'Workflows cannot be executed without the user context',
-            },
-          });
-        }
+        const taskInstance = createTaskInstance(workflowExecution, ['workflows']);
+        await plugins.taskManager.schedule(taskInstance, { request: request as KibanaRequest });
+        this.logger.debug(
+          `Scheduling workflow task with user context for workflow ${workflow.id}, execution ${workflowExecution.id}`
+        );
       }
 
       return {
-        workflowExecutionId: workflowExecution.id,
+        workflowExecutionId: workflowExecution.id as string,
+      };
+    };
+
+    const scheduleWorkflow = async (
+      workflow: WorkflowExecutionEngineModel,
+      context: Record<string, unknown>,
+      request: KibanaRequest
+    ) => {
+      const { workflowExecution } = await createAndPersistWorkflowExecution(
+        workflow,
+        context,
+        'alert'
+      );
+
+      // Always schedule a task (never execute directly)
+      const taskInstance = createTaskInstance(
+        workflowExecution,
+        generateExecutionTaskScope(workflowExecution as EsWorkflowExecution)
+      );
+
+      await plugins.taskManager.schedule(taskInstance, { request });
+      this.logger.debug(
+        `Scheduling workflow task with user context for workflow ${workflow.id}, execution ${workflowExecution.id}`
+      );
+
+      return {
+        workflowExecutionId: workflowExecution.id as string,
       };
     };
 
-    const executeWorkflowStep: ExecuteWorkflowStep = async (workflow, stepId, context, request) => {
+    const executeWorkflowStep = async (
+      workflow: WorkflowExecutionEngineModel,
+      stepId: string,
+      contextOverride: Record<string, unknown>,
+      request?: KibanaRequest | undefined
+    ): Promise<ExecuteWorkflowStepResponse> => {
+      // Check if request is required before creating execution
+      // Workflow steps require user context to run with proper permissions
+      if (!request) {
+        throw new Error('Workflow steps cannot be executed without the user context');
+      }
       await this.initialize(coreStart);
       const workflowCreatedAt = new Date();
 
       // Get ES client and create repository for this execution
       const esClient = coreStart.elasticsearch.client.asInternalUser;
       const workflowExecutionRepository = new WorkflowExecutionRepository(esClient);
+      const context: Record<string, unknown> = {
+        ...contextOverride,
+      };
 
-      const triggeredBy = context.triggeredBy || 'manual'; // 'manual' or 'scheduled'
-      const workflowExecution = {
+      const triggeredBy = (context.triggeredBy as string | undefined) || 'manual'; // 'manual' or 'scheduled'
+      const workflowExecution: Partial<EsWorkflowExecution> = {
```

Comment:
Consider keeping implicit type instead of `Partial<EsWorkflowExecution>`, this approach requires a lot of casting downstream